node/src/manager/commands/poi.rs

```diff
@@ -0,0 +1,655 @@
+            &table.tablename,
+        )?;
+    }
+    //Compress & Upload
+    walk_and_compress(&entity_dump_directory, &entity_compression_directory)?;
+    upload_file_to_endpoint(
+        &entity_compression_directory,
+        &host,
+        indexer_id,
+        dispute_id,
+        "entities",
+    )?;
+
+    Ok(())
+}
+
+/// Create directories required for compression and upload
+fn create_dump_directories(path: &PathBuf, kind: String) -> Result<(), anyhow::Error> {
+    let dir_path = path.join(format!("{}/raw", kind));
+    fs::create_dir_all(&dir_path)?;
+    let compression_dir = path.join(format!("{}/compressed", kind));
+    fs::create_dir_all(&compression_dir)?;
+
+    Ok(())
+}
+
+/// Walks a directory path for all files and adds them to compressed archive.
+fn walk_and_compress(src_dir: &PathBuf, dst_file: &PathBuf) -> zip::result::ZipResult<()> {
+    if !Path::new(src_dir).is_dir() {
+        return Err(ZipError::FileNotFound);
+    }
+
+    let path = Path::new(dst_file);
+    let file = File::create(&path).unwrap();
+
+    let walkdir = WalkDir::new(src_dir);
+    let it = walkdir.into_iter();
+
+    compress(&mut it.filter_map(|e| e.ok()), src_dir, file)?;
+
+    Ok(())
+}
+
+/// Compress contents of a directory
+fn compress<T>(
+    it: &mut dyn Iterator<Item = DirEntry>,
+    prefix: &PathBuf,
+    writer: T,
+) -> zip::result::ZipResult<()>
+where
+    T: Write + Seek,
+{
+    let mut zip = zip::ZipWriter::new(writer);
+    let options = FileOptions::default()
+        .compression_method(zip::CompressionMethod::Deflated)
+        .unix_permissions(0o755);
+
+    let mut buffer = Vec::new();
+    for entry in it {
+        let path = entry.path();
+        let name = path.strip_prefix(Path::new(prefix)).unwrap();
+
+        // Write file or directory explicitly
+        if path.is_file() {
+            println!("Compressing file {:?} as {:?} ...", path, name);
+            #[allow(deprecated)]
+            zip.start_file_from_path(name, options)?;
+            let mut f = File::open(path)?;
+
+            f.read_to_end(&mut buffer)?;
+            zip.write_all(&*buffer)?;
+            buffer.clear();
+        } else if name.as_os_str().len() != 0 {
+            // Only if not root! Avoids path spec / warning
+            // and mapname conversion failed error on unzip
+            println!("Compressing directory {:?} as {:?} ...", path, name);
+            #[allow(deprecated)]
+            zip.add_directory_from_path(name, options)?;
+        }
+    }
+    zip.finish()?;
+    Result::Ok(())
+}
+
+/// Remove all files in the directory after upload
+fn delete_directory(directory: &PathBuf, debug: bool) -> Result<(), anyhow::Error> {
+    if !debug {
+        fs::remove_dir_all(directory)?;
+    }
+    Ok(())
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct PoiRecord {
+    #[sql_type = "Text"]
+    digest: String,
+    #[sql_type = "Text"]
+    id: String,
+    #[sql_type = "BigInt"]
+    vid: i64,
```

Comment:
The `vid` is generally meaningless outside of a specific `graph-node` installations, and can depend on things like how many/which reverts that `graph-node` has seen. It's not useful to report that externally or for comparison.